### PR TITLE
forward auth bearer token

### PR DESCRIPTION
**Pull request by krushanbauva**
v3-engine [PR](https://github.com/hasura/v3-engine/pull/76)

### CODE REVIEW

**Review by paf31 — DISMISSED** [review]

**Comment by paf31** [comment]
@sordina Hopefully this works with what you have planned for hub too, can you please take a look?

@krushanbauva This looks okay to me but I can't test it easily. If we can do an end-to-end test with a hub connector then I can merge this tomorrow. Or feel free to merge it once you've done such a test.

File: ndc-client/src/apis/default_api.rs [hunk 1]
```diff
@@ -61,6 +61,13 @@ pub async fn capabilities_get(
                     .header(reqwest::header::USER_AGENT, user_agent.clone());
             }
 
+            if let Some(ref bearer_token) = configuration.bearer_access_token {
+                req_builder = req_builder.header(
+                    reqwest::header::AUTHORIZATION,
+                    "Bearer ".to_owned() + bearer_token.as_str(),
+                );
```
Comment (paf31) [inline ndc-client/src/apis/default_api.rs:68]: Consider

```suggestion
                req_builder = req_builder.bearer_auth(bearer_token.as_str());
```

See https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.bearer_auth

**Comment by krushanbauva** [comment]
I have tested this with the reference agent and it is working as expected (i.e. it is sending the configured `Authorization` header)
@sordina I will need some help if we want to test this with a hub connector.

**Comment by krushanbauva** [comment]
I was able to test it out end to end locally using a sendgrid connector with the help of @sordina and it works as expected.
The only thing that's remaining is when the request is `unauthorized`, it fails to deserialize the response into a JSON value [here](https://github.com/hasura/ndc-spec/blob/fde4c88b7f0bd02d0bea9ec537991079930cdb01/ndc-client/src/apis/default_api.rs#L173) and hence the error reporting is not accurate.
Currently, when the request is unauthorized, the error we get is
```json
{
  "data": null,
  "errors": [
    {
      "message": "internal communication error: request to connector failed",
      "path": null
    }
  ]
}
```
This can be improved to indicate that the request is unauthorized but I think this needs to be done on the connector side so it sends a valid JSON.

**Review by paf31 — APPROVED** [review]

**Comment by paf31** [comment]
> This can be improved to indicate that the request is unauthorized but I think this needs to be done on the connector side so it sends a valid JSON response in case of unauthorized request.

Right, I think it's this issue: https://github.com/hasura/ndc-hub/pull/27/files#r1327530869

**Pull request merged**